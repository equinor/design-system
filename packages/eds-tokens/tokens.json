{
  "eds": {
    "clickbounds": {
      "jumbo__base": "88px",
      "default__base": "48px",
      "default__input": "56px",
      "compact__standard": "32px"
    },
    "colors": {
      "text": {
        "static_icons__default": "#3d3d3d",
        "static_icons__secondary": "#565656",
        "static_icons__tertiary": "#6f6f6f",
        "static_icons__primary_white": "#ffffff"
      },
      "ui": {
        "background__default": "#ffffff",
        "background__semitransparent": "#ffffff",
        "background__light": "#f7f7f7",
        "background__scrim": "#000000",
        "background__overlay": "#000000",
        "background__medium": "#dcdcdc",
        "background__info": "#d5eaf4",
        "background__warning": "#ffe7d6",
        "background__danger": "#ffc1c1"
      }
    },
    "infographic": {
      "substitute__purple_berry": "#8c1159",
      "substitute__pink_rose": "#e24973",
      "substitute__pink_salmon": "#ff92a8",
      "substitute__green_cucumber": "#005f57",
      "substitute__green_succulent": "#00977b",
      "substitute__green_mint": "#40d38f",
      "substitute__blue_ocean": "#004088",
      "substitute__blue_overcast": "#0084c4",
      "substitute__blue_sky": "#52c0ff",
      "primary__moss_green_100": "#007079",
      "primary__moss_green_55": "#73b1b5",
      "primary__moss_green_34": "#a8ced1",
      "primary__moss_green_21": "#c9e0e2",
      "primary__moss_green_13": "#deedee",
      "primary__energy_red_100": "#eb0037",
      "primary__energy_red_55": "#ff7d98",
      "primary__energy_red_34": "#ffaebf",
      "primary__energy_red_21": "#ffcdd7",
      "primary__energy_red_13": "#ffe0e7",
      "primary__weathered_red": "#7d0023",
      "primary__slate_blue": "#243746",
      "primary__spruce_wood": "#ffe7d6",
      "primary__mist_blue": "#d5eaf4",
      "primary__lichen_green": "#e6faec"
    },
    "logo": {
      "fill_positive": "#eb0037",
      "fill_negative": "#ffffff"
    },
    "interactive": {
      "primary__selected_highlight": "#e6faec",
      "primary__selected_hover": "#c3f3d2",
      "primary__hover_alt": "#deedee",
      "secondary__highlight": "#d5eaf4",
      "secondary__resting": "#243746",
      "secondary__link_hover": "#17242f",
      "danger__highlight": "#ffc1c1",
      "danger__resting": "#eb0000",
      "danger__hover": "#b30d2f",
      "danger__text": "#b30d2f",
      "warning__highlight": "#ffe7d6",
      "warning__resting": "#ff9200",
      "warning__hover": "#ad6200",
      "warning__text": "#ad6200",
      "success__highlight": "#e6faec",
      "success__resting": "#4bb748",
      "success__hover": "#358132",
      "success__text": "#358132",
      "table__cell__fill_resting": "#ffffff",
      "table__cell__fill_hover": "#eaeaea",
      "table__cell__fill_activated": "#e6faec",
      "table__header__fill_activated": "#eaeaea",
      "table__header__fill_hover": "#dcdcdc",
      "table__header__fill_resting": "#f7f7f7",
      "disabled__text": "#bebebe",
      "text_highlight": "#d5eaf4",
      "focus": "#007079",
      "disabled__border": "#dcdcdc",
      "disabled__fill": "#eaeaea",
      "link_on_interactive_colors": "#ffffff",
      "icon_on_interactive_colors": "#ffffff",
      "link_in_snackbars": "#97cace",
      "pressed_overlay_dark": "#000000",
      "pressed_overlay_light": "#ffffff"
    },
    "elevation": {
      "raised": "0 1px 5px rgba(0, 0, 0, 0.2),0 3px 4px rgba(0, 0, 0, 0.12),0 2px 4px rgba(0, 0, 0, 0.14)",
      "none": "0 0 1px rgba(0, 0, 0, 0.14)",
      "overlay": "0 1px 10px rgba(0, 0, 0, 0.2),0 4px 5px rgba(0, 0, 0, 0.12),0 2px 4px rgba(0, 0, 0, 0.14)",
      "sticky": "0 4px 5px rgba(0, 0, 0, 0.2),0 3px 14px rgba(0, 0, 0, 0.12),0 8px 10px rgba(0, 0, 0, 0.14)",
      "temporary_nav": "0 7px 8px rgba(0, 0, 0, 0.2),0 5px 22px rgba(0, 0, 0, 0.12),0 12px 17px rgba(0, 0, 0, 0.14)",
      "above_scrim": "0 11px 15px rgba(0, 0, 0, 0.2),0 9px 46px rgba(0, 0, 0, 0.12),0 24px 38px rgba(0, 0, 0, 0.14)"
    },
    "interactions": {
      "_modes": {
        "compact": {
          "pressed_dark_overlay": {
            "blendMode": "pass_through",
            "pressedColor": "transparent"
          },
          "focused": {
            "style": "dashed",
            "color": "rgba(0, 112, 121, 1)",
            "width": "2px"
          },
          "pressed_light_overlay": {
            "blendMode": "pass_through",
            "pressedColor": "rgba(255, 255, 255, 1)"
          }
        }
      },
      "pressed_dark_overlay": {
        "blendMode": "darken",
        "pressedColor": "rgba(0, 0, 0, 0.2)"
      },
      "focused": {
        "style": "dashed",
        "color": "rgba(0, 112, 121, 1)",
        "width": "2px"
      },
      "pressed_light_overlay": {
        "blendMode": "pass_through",
        "pressedColor": "rgba(255, 255, 255, 0.2)"
      }
    },
    "shape": {
      "_modes": {
        "compact": {
          "rounded": {
            "minHeight": "24px",
            "minWidth": "80px",
            "borderRadius": "100px"
          },
          "button": {
            "minHeight": "24px",
            "minWidth": "112px",
            "borderRadius": "4px"
          },
          "toggle": {
            "minHeight": "24px",
            "minWidth": "112px",
            "borderRadius": "4px"
          },
          "corners": {
            "minHeight": "24px",
            "minWidth": "112px",
            "borderRadius": "4px"
          },
          "icon_button": {
            "minHeight": "32px",
            "minWidth": "32px",
            "borderRadius": "100px"
          },
          "straight": {
            "minHeight": "24px",
            "minWidth": "112px",
            "borderRadius": ""
          },
          "toggle_rounded_border": {
            "minHeight": "24px",
            "minWidth": "2px",
            "borderRadius": ""
          },
          "toggle_straight_border": {
            "minHeight": "24px",
            "minWidth": "2px",
            "borderRadius": ""
          },
          "circle": {
            "minHeight": "24px",
            "minWidth": "24px",
            "borderRadius": "100px"
          }
        }
      },
      "toggle_straight_border": {
        "minHeight": "36px",
        "minWidth": "4px",
        "borderRadius": ""
      },
      "toggle_rounded_border": {
        "minHeight": "36px",
        "minWidth": "4px",
        "borderRadius": ""
      },
      "circle": {
        "minHeight": "40px",
        "minWidth": "40px",
        "borderRadius": "100px"
      },
      "rounded": {
        "minHeight": "40px",
        "minWidth": "80px",
        "borderRadius": "100px"
      },
      "button": {
        "minHeight": "36px",
        "minWidth": "112px",
        "borderRadius": "4px"
      },
      "toggle": {
        "minHeight": "36px",
        "minWidth": "112px",
        "borderRadius": "4px"
      },
      "corners": {
        "minHeight": "36px",
        "minWidth": "112px",
        "borderRadius": "4px"
      },
      "icon_button": {
        "minHeight": "40px",
        "minWidth": "40px",
        "borderRadius": "100px"
      },
      "field": {
        "minHeight": "56px",
        "minWidth": "294px",
        "borderRadius": ""
      },
      "straight": {
        "minHeight": "36px",
        "minWidth": "112px",
        "borderRadius": ""
      },
      "caret": {
        "minHeight": "8px",
        "minWidth": "6px",
        "borderRadius": ""
      }
    },
    "typography": {
      "heading": {
        "h1_bold": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "2.000rem",
          "fontWeight": 700,
          "lineHeight": "1.250em",
          "textAlign": "left"
        },
        "h1": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "2.000rem",
          "fontWeight": 400,
          "lineHeight": "1.250em",
          "textAlign": "left"
        },
        "h2": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.750rem",
          "fontWeight": 400,
          "lineHeight": "1.250em",
          "textAlign": "left"
        },
        "h3": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.500rem",
          "fontWeight": 400,
          "lineHeight": "1.250em",
          "textAlign": "left"
        },
        "h4": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.250rem",
          "fontWeight": 400,
          "lineHeight": "1.600em",
          "textAlign": "left"
        },
        "h5": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.125rem",
          "fontWeight": 500,
          "letterSpacing": "0.013em",
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "h6": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 500,
          "letterSpacing": "0.013em",
          "lineHeight": "1.500em",
          "textAlign": "left"
        }
      },
      "navigation": {
        "menu_title": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "letterSpacing": "0.013em",
          "lineHeight": "1.000em",
          "textAlign": "left"
        },
        "menu_tabs": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 500,
          "letterSpacing": "0.013em",
          "lineHeight": "1.000em",
          "textAlign": "left"
        },
        "label": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "drawer_active": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 700,
          "letterSpacing": "0.006em",
          "lineHeight": "1.000em",
          "textAlign": "left"
        },
        "drawer_inactive": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 500,
          "letterSpacing": "0.013em",
          "lineHeight": "1.000em",
          "textAlign": "left"
        },
        "button": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.143em",
          "textAlign": "left"
        },
        "breadcrumb": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.143em",
          "textAlign": "left"
        },
        "breadcrumb_hover": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.143em",
          "textDecoration": "underline",
          "textAlign": "left"
        },
        "menu_title_hover": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "letterSpacing": "0.013em",
          "lineHeight": "1.000em",
          "textAlign": "left"
        }
      },
      "input": {
        "label": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "text": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "letterSpacing": "0.025em",
          "lineHeight": "1.500em",
          "textAlign": "left"
        },
        "text_monospaced": {
          "fontFeature": "'tnum' on,'lnum' on",
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "letterSpacing": "0.063em",
          "lineHeight": "1.500em",
          "textAlign": "left"
        },
        "helper": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "letterSpacing": "0.013em",
          "lineHeight": "1.333em",
          "textAlign": "left"
        }
      },
      "paragraph": {
        "body_short_italic": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "lineHeight": "1.250em",
          "fontStyle": "italic",
          "textAlign": "left"
        },
        "caption": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.143em",
          "textAlign": "left"
        },
        "meta": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.625rem",
          "fontWeight": 500,
          "lineHeight": "1.600em",
          "textAlign": "left"
        },
        "body_short": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "lineHeight": "1.250em",
          "textAlign": "left"
        },
        "body_short_bold_italic": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 700,
          "lineHeight": "1.250em",
          "fontStyle": "italic",
          "textAlign": "left"
        },
        "body_short_bold": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 700,
          "lineHeight": "1.250em",
          "textAlign": "left"
        },
        "body_short_link": {
          "color": "rgba(0, 112, 121, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "lineHeight": "1.250em",
          "textDecoration": "underline",
          "textAlign": "left"
        },
        "overline": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.625rem",
          "fontWeight": 500,
          "letterSpacing": "0.069em",
          "lineHeight": "1.600em",
          "textTransform": "uppercase",
          "textAlign": "left"
        },
        "ingress": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.125rem",
          "fontWeight": 400,
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "body_long": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "lineHeight": "1.500em",
          "textAlign": "left"
        },
        "body_long_link": {
          "color": "rgba(0, 112, 121, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "lineHeight": "1.500em",
          "textDecoration": "underline",
          "textAlign": "left"
        },
        "body_long_italic": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 400,
          "lineHeight": "1.500em",
          "fontStyle": "italic",
          "textAlign": "left"
        },
        "body_long_bold": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 700,
          "lineHeight": "1.500em",
          "textAlign": "left"
        },
        "body_long_bold_italic": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 700,
          "lineHeight": "1.500em",
          "fontStyle": "italic",
          "textAlign": "left"
        }
      },
      "table": {
        "cell_header": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 700,
          "lineHeight": "1.429em",
          "textAlign": "left"
        },
        "cell_text": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.429em",
          "textAlign": "left"
        },
        "cell_text_bold": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 700,
          "lineHeight": "1.429em",
          "textAlign": "left"
        },
        "cell_text_link": {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.429em",
          "textDecoration": "underline",
          "textAlign": "left"
        },
        "cell_numeric_monospaced": {
          "fontFeature": "'tnum' on,'lnum' on",
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.429em",
          "textAlign": "left"
        }
      },
      "_modes": {
        "compact": {
          "table": {
            "cell_header": {
              "color": "rgba(0, 0, 0, 1)",
              "fontFamily": "Equinor",
              "fontSize": "0.875rem",
              "fontWeight": 700,
              "lineHeight": "1.143em",
              "textAlign": "left"
            },
            "cell_text": {
              "color": "rgba(0, 0, 0, 1)",
              "fontFamily": "Equinor",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "lineHeight": "1.143em",
              "textAlign": "left"
            },
            "cell_text_bold": {
              "color": "rgba(0, 0, 0, 1)",
              "fontFamily": "Equinor",
              "fontSize": "0.875rem",
              "fontWeight": 700,
              "lineHeight": "1.143em",
              "textAlign": "left"
            },
            "cell_text_link": {
              "color": "rgba(0, 0, 0, 1)",
              "fontFamily": "Equinor",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "lineHeight": "1.143em",
              "textDecoration": "underline",
              "textAlign": "left"
            },
            "cell_numeric_monospaced": {
              "fontFeature": "'tnum' on,'lnum' on",
              "color": "rgba(0, 0, 0, 1)",
              "fontFamily": "Equinor",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "lineHeight": "1.143em",
              "textAlign": "left"
            }
          }
        }
      },
      "ui": {
        "tooltip": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "snackbar": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "accordion_header": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "1.000rem",
          "fontWeight": 500,
          "lineHeight": "1.500em",
          "textAlign": "left"
        },
        "chip__badge": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "lineHeight": "1.333em",
          "textAlign": "left"
        },
        "chart": {
          "color": "rgba(61, 61, 61, 1)",
          "fontFamily": "Equinor",
          "fontSize": "0.750rem",
          "fontWeight": 500,
          "lineHeight": "1.333em",
          "textAlign": "left"
        }
      }
    },
    "spacings": {
      "comfortable": {
        "xxx_large": "48px",
        "xx_large": "40px",
        "x_large": "32px",
        "large": "24px",
        "medium": "16px",
        "medium_small": "12px",
        "small": "8px",
        "x_small": "4px",
        "xx_small": "2px"
      }
    }
  }
}
