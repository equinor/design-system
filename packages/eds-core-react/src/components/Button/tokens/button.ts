import { tokens } from '@equinor/eds-tokens'
import { ButtonToken } from '../Button.types'

const {
  typography: {
    navigation: { button: buttonTypography },
  },
  colors: {
    interactive: {
      disabled__text: { rgba: disabledTextColor },
      disabled__border: { rgba: disabledBorderColor },
      disabled__fill: { rgba: disabledColor },
    },
  },
  spacings: {
    comfortable: { medium },
  },
  shape: {
    button: { minHeight: buttonHeight, borderRadius: buttonBorderRadius },
    _modes: {
      compact: {
        button: { minHeight: compactButtonHeight },
      },
    },
  },
  interactions: { focused: outline },
  clickbounds: {
    default__base: clicboundHeight,
    compact__standard: compactClickboundHeight,
  },
} = tokens

export const button: ButtonToken = {
  background: 'transparent',
  height: `var(--eds_button__height, ${buttonHeight})`,
  typography: {
    ...buttonTypography,
    textAlign: 'center',
    fontSize: `var(--eds_button__font_size, ${buttonTypography.fontSize})`,
  },
  border: {
    type: 'border',
    width: `var(--eds_button__border_width, 1px)`,
    color: 'transparent',
    radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
    style: 'solid',
  },
  spacings: {
    top: 'var(--eds_button__padding_y, 0)',
    bottom: 'var(--eds_button__padding_y, 0)',
    left: `var(--eds_button__padding_x, ${medium})`,
    right: `var(--eds_button__padding_x, ${medium})`,
  },
  clickbound: {
    height: clicboundHeight,
    width: '100%',
    offset: {
      top: `${(parseInt(clicboundHeight) - parseInt(buttonHeight)) / 2 + 1}px`,
      left: '0',
    },
  },
  entities: {
    icon: {
      height: 'var(--eds_button__icon__size, 24px)',
      width: 'var(--eds_button__icon__size, 24px)',
    },
  },
  states: {
    hover: {
      border: {
        type: 'border',
        width: '1px',
        color: 'transparent',
        radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
        style: 'solid',
      },
    },
    focus: {
      outline: {
        type: 'outline',
        offset: '2px',
        style: 'dashed',
        color: outline.color,
        width: outline.width,
      },
    },
    disabled: {
      background: disabledColor,
      border: {
        type: 'border',
        width: '1px',
        color: disabledBorderColor,
        style: 'solid',
      },
      typography: {
        ...buttonTypography,
        color: disabledTextColor,
        textAlign: 'center',
      },
    },
  },
  modes: {
    compact: {
      height: `var(--eds_button__height_compact, ${compactButtonHeight})`,
      spacings: {
        top: 'var(--eds_button__padding_y_compact, 0)',
        bottom: 'var(--eds_button__padding_y_compact, 0)',
      },
      clickbound: {
        height: compactClickboundHeight,
        width: '100%',
        offset: {
          top: `${
            (parseInt(compactClickboundHeight) -
              parseInt(compactButtonHeight)) /
              2 +
            1
          }px`,
          left: '0',
        },
      },
    },
  },
}
