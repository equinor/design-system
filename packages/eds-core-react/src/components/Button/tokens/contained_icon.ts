import { tokens } from '@equinor/eds-tokens'
import { mergeDeepRight } from 'ramda'

import {
  primary as primaryContained,
  secondary as secondaryContained,
  danger as dangerContained,
} from './contained'
import { ButtonToken } from '../Button.types'

const {
  clickbounds: {
    default__base: clicboundHeight,
    compact__standard: compactClickboundHeight,
  },
  shape,
} = tokens

const contained_icon = {
  height: shape.icon_button.minHeight,
  width: shape.icon_button.minWidth,
  border: {
    width: '0px',
    radius: '50%',
  },
  spacings: { left: '0', right: '0' },
  clickbound: {
    width: clicboundHeight,
    offset: {
      top: `${
        (parseInt(clicboundHeight) - parseInt(shape.icon_button.minWidth)) / 2
      }px`,
      left: `${
        (parseInt(clicboundHeight) - parseInt(shape.icon_button.minWidth)) / 2
      }px`,
    },
  },
  states: {
    hover: {
      border: {
        width: '0px',
        radius: '50%',
      },
    },
  },
  modes: {
    compact: {
      height: shape._modes.compact.icon_button.minHeight,
      width: shape._modes.compact.icon_button.minWidth,
      clickbound: {
        width: compactClickboundHeight,
        offset: {
          top: '0',
          left: `${
            (parseInt(compactClickboundHeight) -
              parseInt(shape._modes.compact.icon_button.minWidth)) /
            2
          }px`,
        },
      },
    },
  },
}

export const primary = mergeDeepRight(
  primaryContained,
  contained_icon,
) as ButtonToken

export const secondary = mergeDeepRight(
  secondaryContained,
  contained_icon,
) as Partial<ButtonToken>
export const danger = mergeDeepRight(
  dangerContained,
  contained_icon,
) as Partial<ButtonToken>
