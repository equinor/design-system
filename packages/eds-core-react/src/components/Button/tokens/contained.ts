import { tokens } from '@equinor/eds-tokens'
import { ButtonToken } from '../Button.types'
import { button } from './button'

import { mergeDeepRight } from 'ramda'

const {
  colors: {
    text: {
      static_icons__primary_white: { rgba: primaryWhite },
    },
    interactive: {
      primary__resting: { rgba: primaryColor },
      primary__hover: { rgba: primaryHoverColor },
      secondary__resting: { rgba: secondaryColor },
      secondary__link_hover: { rgba: secondaryHoverColor },
      danger__resting: { rgba: dangerColor },
      danger__hover: { rgba: dangerHoverColor },
    },
  },

  shape: {
    button: { borderRadius: buttonBorderRadius },
  },
} = tokens

export const primary = mergeDeepRight(button, {
  background: primaryColor,
  typography: {
    color: primaryWhite,
  },
  border: {
    type: 'border',
    style: 'solid',
    width: '1px',
    color: primaryColor,
    radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
  },
  states: {
    hover: {
      background: primaryHoverColor,
    },
  },
}) as ButtonToken
export const secondary = mergeDeepRight(primary, {
  background: secondaryColor,
  border: {
    color: secondaryColor,
  },
  states: {
    hover: {
      background: secondaryHoverColor,
      border: {
        color: secondaryHoverColor,
      },
    },
  },
}) as ButtonToken

export const danger = mergeDeepRight(primary, {
  background: dangerColor,
  border: {
    color: dangerColor,
  },
  states: {
    hover: {
      background: dangerHoverColor,
      border: {
        color: dangerHoverColor,
      },
    },
  },
}) as ButtonToken
