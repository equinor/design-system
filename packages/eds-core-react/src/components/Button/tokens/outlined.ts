import { tokens } from '@equinor/eds-tokens'
import { mergeDeepRight } from 'ramda'
import { button } from './button'
import { ButtonToken } from '../Button.types'

const {
  colors: {
    interactive: {
      primary__resting: { rgba: primaryColor },
      primary__hover: { rgba: primaryHoverColor },
      primary__hover_alt: { rgba: primaryHoverAltColor },
      secondary__resting: { rgba: secondaryColor },
      secondary__link_hover: { rgba: secondaryHoverColor },
      secondary__highlight: { rgba: secondaryHoverAltColor },
      danger__resting: { rgba: dangerColor },
      danger__hover: { rgba: dangerHoverColor },
      danger__highlight: { rgba: dangerHoverAltColor },
    },
  },

  shape: {
    button: { borderRadius: buttonBorderRadius },
  },
} = tokens

export const primary = mergeDeepRight(button, {
  typography: {
    color: primaryColor,
  },
  border: {
    type: 'border',
    style: 'solid',
    width: '1px',
    color: primaryColor,
    radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
  },
  states: {
    hover: {
      typography: {
        color: primaryHoverColor,
      },
      background: primaryHoverAltColor,
      border: {
        style: 'solid',
        type: 'border',
        width: '1px',
        color: primaryHoverColor,
      },
    },
    disabled: {
      background: 'transparent',
    },
  },
}) as ButtonToken

export const secondary = mergeDeepRight(primary, {
  typography: {
    color: secondaryColor,
  },
  border: {
    color: secondaryColor,
  },
  states: {
    hover: {
      background: secondaryHoverAltColor,
      typography: {
        color: secondaryHoverColor,
      },
      border: {
        color: secondaryHoverColor,
      },
    },
  },
}) as ButtonToken

export const danger = mergeDeepRight(primary, {
  typography: {
    color: dangerColor,
  },
  border: {
    color: dangerColor,
  },
  states: {
    hover: {
      background: dangerHoverAltColor,
      typography: {
        color: dangerHoverColor,
      },
      border: {
        color: dangerHoverColor,
      },
    },
  },
}) as ButtonToken
