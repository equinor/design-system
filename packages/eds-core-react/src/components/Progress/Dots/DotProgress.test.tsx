/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import * as tokens from './DotProgress.tokens'
import { DotProgress } from './DotProgress'

const StyledProgress = styled(DotProgress)`
  position: absolute;
`

describe('DotProgress', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(<DotProgress />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(<DotProgress />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('should render neutral as default', () => {
    render(<DotProgress />)
    const progressbar = screen.getByRole('progressbar')
    expect(progressbar).toHaveStyleRule('fill', tokens.neutral.background)
  })
  it('should render primary color if stated in props', () => {
    render(<DotProgress color="primary" />)
    const progressbar = screen.getByRole('progressbar')
    expect(progressbar).toHaveStyleRule('fill', tokens.primary.background)
  })
  it('can extend the css for the component', () => {
    render(<StyledProgress />)
    const progressbar = screen.getByRole('progressbar')
    expect(progressbar).toHaveStyleRule('position', 'absolute')
  })
  it('has correct default size ', () => {
    render(<StyledProgress />)
    const progressbar = screen.getByRole('progressbar')

    expect(progressbar).toHaveAttribute('width', '32')
    expect(progressbar).toHaveAttribute('height', `${32 / 4}`)
  })
  it('has correct size when size is 64', () => {
    render(<StyledProgress size={64} />)
    const progressbar = screen.getByRole('progressbar')

    expect(progressbar).toHaveAttribute('width', '64')
    expect(progressbar).toHaveAttribute('height', `${64 / 4}`)
  })
})
