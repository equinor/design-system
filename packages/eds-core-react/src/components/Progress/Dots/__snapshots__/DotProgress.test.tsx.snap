// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotProgress Matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  fill: var(--eds_interactive__icon_on_interactive_colors, rgba(255, 255, 255, 1));
}

.c0 circle:nth-child(1) {
  animation: jkwjQg 1s ease infinite;
}

.c0 circle:nth-child(2) {
  animation: jkwjQg 1s ease 0.2s infinite;
}

.c0 circle:nth-child(3) {
  animation: jkwjQg 1s ease 0.4s infinite;
}

<svg
    class="c0"
    color="var(--eds_interactive__icon_on_interactive_colors, rgba(255, 255, 255, 1))"
    height="8"
    preserveAspectRatio="xMidYMid meet"
    role="progressbar"
    viewBox="0 0 16 4"
    width="32"
  >
    <circle
      cx="2"
      cy="2"
      r="2"
    />
    <circle
      cx="8"
      cy="2"
      r="2"
    />
    <circle
      cx="14"
      cy="2"
      r="2"
    />
  </svg>
</DocumentFragment>
`;
